dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/AnonymisingDataService.java

```diff
@@ -0,0 +1,281 @@
+    }
+
+    private CONTRIBUTIONS.CapitalSummary anonymiseCapitalSummary(CONTRIBUTIONS.CapitalSummary capitalSummary) {
+
+        Optional.ofNullable(capitalSummary.getMotorVehicleOwnership()).ifPresent(motorVehicleOwnership -> {
+            if (hasValue(motorVehicleOwnership.getOwner())) {
+                motorVehicleOwnership.setOwner(faker.name().fullName());
+            }
+            Optional.ofNullable(motorVehicleOwnership.getRegistrationList()).ifPresent(registrationList -> {
+                if (hasValue(registrationList.getRegistration())) {
+                    registrationList.setRegistration(faker.vehicle().licensePlate());
+                }
+            });
+        });
+        return capitalSummary;
+    }
+
+    private CONTRIBUTIONS.Application anonymiseApplication(CONTRIBUTIONS.Application application) {
+
+        if (hasValue(application.getArrestSummonsNumber())) {
+            application.setArrestSummonsNumber(faker.number().digits(5));
+        }
+        return application;
+    }
+
+    private CONTRIBUTIONS.Applicant anonymiseApplicant(CONTRIBUTIONS.Applicant applicant) {
+
+        anonymisePersonalDetails(applicant);
+        Optional.ofNullable(applicant.getBankDetails()).ifPresent(this::anonymiseBankDetails);
+        Optional.ofNullable(applicant.getPartnerDetails()).ifPresent(this::anonymisePartnerPersonalDetails);
+        Optional.ofNullable(applicant.getPartner()).ifPresent(this::anonymisePartnerCirDetails);
+        Optional.ofNullable(applicant.getDisabilitySummary()).ifPresent(this::anonymiseDisabilitySummary);
+        Optional.ofNullable(applicant.getHomeAddress()).ifPresent(this::anonymizeHomeAddress);
+        Optional.ofNullable(applicant.getPostalAddress()).ifPresent(this::anonymizePostalAddress);
+
+        return applicant;
+    }
+
+
+    private void anonymisePersonalDetails(CONTRIBUTIONS.Applicant applicant) {
+        Optional.ofNullable(applicant.getId()).ifPresent(id -> applicant.setId(faker.number().positive()));
+        if (hasValue(applicant.getFirstName())) {
+            applicant.setFirstName(faker.name().firstName());
+        }
+        if (hasValue(applicant.getLastName())) {
+            applicant.setLastName(faker.name().lastName());
+        }
+        Optional.ofNullable(applicant.getDob()).ifPresent(dob ->
+                applicant.setDob(convertToXMLGregorianCalendar(faker.timeAndDate().birthday()))
+        );
+        if (hasValue(applicant.getNiNumber())) {
+            applicant.setNiNumber(faker.idNumber().valid());
+        }
+        if (hasValue(applicant.getLandline())) {
+            applicant.setLandline(faker.phoneNumber().phoneNumber());
+        }
+        if (hasValue(applicant.getMobile())) {
+            applicant.setMobile(faker.phoneNumber().cellPhone());
+        }
+        if (hasValue(applicant.getEmail())) {
+            applicant.setEmail(faker.internet().emailAddress());
+        }
+        if (hasValue(applicant.getSpecialInvestigation())) {
```

Comment:
hi I know the confluence page says to anonymise this but it shouldn't be. Can you remove it from the code ? I will update the confluence page

Reply:
ps this was for the 'special investigation' field